extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -668,13 +714,38 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
         };
       }
     }
+    return null;
+  }
 
-    // Just display in the active viewport
-    return {
-      viewportId: activeViewportId,
-      displaySetInstanceUID,
-      viewportOptions: { viewportType },
-    };
+  /**
+   * Use a viewport with matching orientation (IOP) when no direct displaySet match is found.
+   */
+  private _findViewportWithMatchingOrientation(displaySetInstanceUID, viewportType) {
+    const { displaySetService } = this.servicesManager.services;
+
+    const closestOrientation = getClosestOrientationFromIOP(
```

Comment:
There is another PR coming up that sorts the viewports by closest orientation - can we wait on your PR until that PR merges, and then re-use the same logic (in this class even) to get the closest one possible rather than trying for an exact match?

Reply:
https://github.com/OHIF/Viewers/pull/5090 is the PR I'd like to share code with.